tests/validations.sh

```diff
@@ -288,15 +288,54 @@ spk service install-build-pipeline --org-name $AZDO_ORG -u $remote_repo_url -d $
 pipeline_created=$(az pipelines show --name $frontend_pipeline_name --org $AZDO_ORG_URL --p $AZDO_PROJECT)
 
 # Verify frontend service pipeline run was successful
-verify_pipeline_with_poll $AZDO_ORG_URL $AZDO_PROJECT $frontend_pipeline_name 300 15
+verify_pipeline_with_poll $AZDO_ORG_URL $AZDO_PROJECT $frontend_pipeline_name 300 15 1
 
 echo "Finding pull request that $frontend_pipeline_name pipeline created..."
 approve_pull_request $AZDO_ORG_URL $AZDO_PROJECT "Updating $FrontEnd image tag to master"
 # --------------------------------
 
+##################################
+# App Mono Repo create ring
+##################################
+ring_name=test
+
+cd $TEST_WORKSPACE
+cd $mono_repo_dir
+
+echo "Create ring"
+spk ring create $ring_name
+git add -A
+git commit -m "Adding test ring"
+git push -u origin --all
+
+# Wait for the lifecycle pipeline to finish and approve the pull request
+verify_pipeline_with_poll $AZDO_ORG_URL $AZDO_PROJECT $lifecycle_pipeline_name 180 15 2
+echo "Finding pull request that $lifecycle_pipeline_name pipeline created..."
+approve_pull_request $AZDO_ORG_URL $AZDO_PROJECT "Reconciling HLD"
+
+# Wait for fabrikam-hld-to-fabrikam-manifests pipeline to finish
+verify_pipeline_with_poll $AZDO_ORG_URL $AZDO_PROJECT $hld_to_manifest_pipeline_name 300 15 4
+
+# Verify the file was added in the manifest repository
+cd $TEST_WORKSPACE
+cd $manifests_dir
+
+git pull origin master
+
+ring_dir="prod/$mono_repo_dir/fabrikam-acme-frontend/test"
+if [ ! -d "$ring_dir" ]; then
+  echo "Directory '$ring_dir' does not exist"
+  exit 1
+fi
+
```

Comment:
Because we know the expected content of the rings directory, can we at least check the generated ingressRoutes in `static.yaml`?

sample for master:
```
---
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: fabrikam-acme-frontend-master
spec:
  routes:
    - kind: Rule
      match: 'PathPrefix(`/fabrikam-acme-frontend`) && Headers(`Ring`, `master`)'
      middlewares:
        - name: fabrikam-acme-frontend-master
      services:
        - name: fabrikam-acme-frontend-master
          port: 80

---
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: fabrikam-acme-frontend-master
spec:
  stripPrefix:
    forceSlash: false
    prefixes:
      - /fabrikam-acme-frontend
```

Can we validate the names and the routing rules?

`match: 'PathPrefix(`/fabrikam-acme-frontend`) && Headers(`Ring`, `master`)'`